scripts/azureml-assets/azureml/assets/release/asset_publish.py

```diff
@@ -231,49 +232,58 @@ def _str2bool(v: str) -> bool:
     logger.print(f"create list: {publish_list}")
 
     failure_list = []
-    for asset in util.find_assets(input_dirs=assets_dir):
-        asset_names = publish_list.get(asset.type.value, [])
-        if not ("*" in asset_names or asset.name in asset_names):
-            logger.print(
-                f"Skipping registering asset {asset.name} because it is not in the publish list")
+    all_assets = util.find_assets(input_dirs=assets_dir)
+    assets_by_type = {}
+    for asset in all_assets:
+        assets_by_type[asset.type.value] = assets_by_type.get(asset.type.value, []).append(asset)
+
+    for publish_asset_type in PUBLISH_ORDER:
+        logger.print(f"now publishing {publish_asset_type.value}s.")
+        if publish_asset_type.value not in publish_list:
             continue
-        logger.print(f"Registering {asset}")
-        final_version = asset.version + "-" + \
-            passed_version if passed_version else asset.version
-        logger.print(f"final version: {final_version}")
-        asset_ids[asset.name] = ASSET_ID_TEMPLATE.substitute(
-            registry_name=registry_name,
-            asset_type=f"{asset.type.value}s",
-            asset_name=asset.name,
-            version=final_version,
-        )
+        for asset in assets_by_type.get(publish_asset_type.value, []):
```

Comment:
```suggestion
        for asset in assets_by_type.get(publish_asset_type, []):
```

Reply:
I think we should have `.value` since we are using lower cases in keys

Reply:
If you use enum values as keys in `assets_by_type` you just need to be consistent in setting and reading them. This part below uses the string value against the YAML that was read in: `if not ("*" in asset_names or asset.name in asset_names):`

Reply:
I think we are always using `.value` when interacting with the `publish_list` since it is using lower cases.

Reply:
changed `assets_by_type`'s keys to string